app/UpdateStores.tsx

```diff
@@ -0,0 +1,99 @@
+import { ReactElement, useState } from "react";
+import { Wrapper } from "./ui/Wrapper";
+import { greenActiveButton, greyInactiveButton } from "./ui/colourLibrary";
+import { LargeButton } from "./ui/LargeButton";
+import { StyledForm } from "./ui/StyledForm";
+import { PageHeader } from "./ui/PageHeader";
+import { FormLabel } from "./ui/FormLabel";
+import { FormGroup } from "./ui/FormGroup";
+import { FormControl } from "./ui/FormControl";
+
+export function UpdateStores(props: {
+  onUpdateStoreClick: () => void;
+  onBackClick: () => void;
+  //   prefilledDisplayName: string;
+  //   prefilledFullName: string;
+  //   prefilledSuburb: string;
+}): ReactElement {
+  const [
+    errorMessage,
+    setErrorMessage,
+    // prefilledDisplayName,
+    // prefilledFullName,
+    // prefilledSuburb,
+  ] = useState("");
+
+  const [displayName, setdisplayName] = useState("");
+  const [fullName, setFullName] = useState("");
+  const [suburb, setSuburb] = useState("");
+
+  const { onUpdateStoreClick, onBackClick } = props;
+  return (
+    <Wrapper>
+      <PageHeader> STORE NAME</PageHeader>
+      <div>
+        <StyledForm>
+          <FormGroup>
+            <FormLabel>Display name</FormLabel>
+            <FormControl
+              type="text"
+              placeholder="enter display name of store"
+              value={displayName}
+              onChange={setdisplayName}
+            />
+          </FormGroup>
+
+          <FormGroup>
+            <FormLabel>Full name</FormLabel>
+
+            <FormControl
+              type="text"
+              placeholder="enter full name of store"
+              value={fullName}
+              onChange={setFullName}
+            />
+          </FormGroup>
+          <FormGroup>
+            <FormLabel>Suburb</FormLabel>
+
+            <FormControl
+              type="text"
+              placeholder="enter suburb"
+              value={suburb}
+              onChange={setSuburb}
+            />
+          </FormGroup>
+        </StyledForm>
+      </div>
+      {errorMessage}
+
+      <div style={{ display: "flex" }}>
+        <LargeButton
+          onClick={() => {
+            onBackClick();
+          }}
```

Comment:
can pass the function directly since it's the only thing in this anonymous callback, ie you can do `onClick={onBackClick}` make sure you understand why this is possible plz

Reply:
here is my explanation:

onClick is a function that receives no arguments and results in void
onBackClick by itself (the instructions) is a function that receives no arguments and results in void
onClick={( ) => onBackClick( ) } is a function that receives no arguments, and results in void after executing the onBackClick function  

so you can't say onClick = {onBackClick( )} because void is not equivalent to a function that results in void 

https://github.com/auberginearms/groceryDatabaseApp/pull/19/commits/5f9b8a3c0bf75637cdfd598656cd9dd647fa230a

Reply:
Yup correct, but you can do what you did which was just pass the instructions straight on